src/simulator/instruction.rs

```diff
@@ -0,0 +1,40 @@
+#![cfg_attr(not(test), allow(dead_code))]
+
+extern crate foxbox_taxonomy;
+
+use foxbox_taxonomy::util::Id;
+use foxbox_taxonomy::values::*;
+use foxbox_taxonomy::devices::*;
+use foxbox_taxonomy::selector::*;
+use foxbox_taxonomy::api::{WatchEvent, WatchOptions, Error};
+
+#[derive(Serialize, Deserialize, Debug)]
+/// Instructions given to the simulator.
+pub enum Instruction {
+    AddNodes(Vec<Node>),
+    AddGetters(Vec<Channel<Getter>>),
+    AddSetters(Vec<Channel<Setter>>),
+    InjectGetterValue{id: Id<Getter>, value: Value},
+}
+
+/// Operations internal to the simulator.
+pub enum Op {
```

Comment:
There is an inconsistency in this module.

If this module is public, then both `Instruction` and `Op` will end up public. We don't want `Op` to be public.
If this module is private, then both `Instruction` and `Op` will be private. We don't want `Instruction` to be private.

Consequently, I believe that we should not put `Instruction` and `Op` in the same module. `Instruction` should end in a `pub mod`, while `Op` should not.